lora-phy/src/sx127x/mod.rs

```diff
@@ -58,6 +58,7 @@ pub struct Config<C: Sx127xVariant> {
 pub struct Sx127x<SPI, IV, C: Sx127xVariant + Sized> {
     intf: SpiInterface<SPI, IV>,
     config: Config<C>,
+    chip_version: Option<u8>,
```

Comment:
Firstly, lets use `sensitivity_quirk: false` as this sensitivity errata seems to be only needed for `sx127{6,7,8}` and not sx1272.

So, we would need to go through chip-specific functions:

```rust
self.sensitivity_quirk = C::is_errata_needed().await?;
```

Where sx1272 function returns just `false` and sx1276 specific version looks something like this:
```rust
{
  let chip_version = self.read_register(Register::RegVersion).await?;
  debug!("Detected sx127x v{}", chip_version);
  chip_version == 0x12
};
```

Reply:
took a slightly different approach to see what you think. I like the use of associated types because it doesn't leak the existence of 1276 implementation details to the 1272 chip, as well as provides a generic way for variants to store additional data if required. However, it does make the types a bit more complicated, and relies on the data type implementing default.